### PR TITLE
Channel

### PR DESCRIPTION
**Pull request by PalletBoy**
Supports GetChannel and CreateChannel

### CODE REVIEW

File: src/AsteriskWrapper/ChannelWrapper.cs [hunk 1]
```diff
@@ -66,5 +66,15 @@ public Task<Playback> StartPlaybackAsync(string media, CancellationToken cancell
         {
             return Channels.StartPlaybackAsync(ChannelId, media, cancellationToken);
         }
+
+        public Task<Channel> GetChannel()
+        {
+            return Channels.GetChannelAsync(ChannelId);
+        }
+
+        public Task<string> CreateChannel(string endpoint, string extension, string context, string priority, string app, CallerId callerId, string label = "", string appArgs = "", int timeout = 30, string channelId = "", string otherChannelId = "", string originator = "")
```
Comment (nufyoot) [inline src/AsteriskWrapper/ChannelWrapper.cs]: That is a LOT of parameters. Can we create an object `CreateChannelOptions` with all these properties and perhaps some constructors to enforce that the bare minimum information is provided?

Comment (nufyoot) [inline src/AsteriskWrapper/ChannelWrapper.cs]: Also, Don't forget your `Async` suffixes :smile:

Comment (nufyoot) [inline src/AsteriskWrapper/ChannelWrapper.cs]: Also, `CreateChannelAsync` doesn't really make sense here. The `ChannelWrapper` was built for wrapping existing channels, so having a `ChannelWrapper` to create a new channel won't work.

File: src/AsteriskWrapper/ChannelWrapper.cs [hunk 2]
```diff
@@ -66,5 +66,15 @@ public Task<Playback> StartPlaybackAsync(string media, CancellationToken cancell
         {
             return Channels.StartPlaybackAsync(ChannelId, media, cancellationToken);
         }
+
+        public Task<Channel> GetChannel()
```
Comment (nufyoot) [inline src/AsteriskWrapper/ChannelWrapper.cs]: `Async` suffix

File: src/AsteriskWrapper/Channels.cs [hunk 3]
```diff
@@ -128,5 +128,54 @@ public Task<Playback> StartPlaybackAsync(string channelId, string media)
...
+            using (var response = await httpClient.GetAsync($"/ari/channels/{channelId}", cancellationToken).ConfigureAwait(false))
+            {
+                if (!response.IsSuccessStatusCode)
+                    throw await response.ToExceptionAsync().ConfigureAwait(false);
+
+                return JsonConvert.DeserializeObject<Channel>(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            }
+        }
+
+        public Task<string> CreateChannel(string endpoint, string extension, string context, string priority, string app, CallerId callerId, string label, string appArgs, int timeout, string channelId, string otherChannelId, string originator)
```
Comment (nufyoot) [inline src/AsteriskWrapper/Channels.cs]: Just as I mentioned inside `ChannelWrapper`, I would rather see a `CreateChannelOptions` class here.

File: src/AsteriskWrapper/ChannelWrapper.cs (hunk 1 above)
Comment (PalletBoy) [inline src/AsteriskWrapper/ChannelWrapper.cs]: `CreateChannelAsync doesn't really make sense here` - well - yes and no... the ARI handles the origination in the same API collection for Channels:
https://wiki.asterisk.org/wiki/display/AST/Asterisk+13+Channels+REST+API#Asterisk13ChannelsRESTAPI-originate

File: src/AsteriskWrapper/OriginateParameters.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            get
+            {
+                return timeout;
+            }
+            set
+            {
+                timeout = value;
+            }
+        }
```
Comment (nufyoot) [inline src/AsteriskWrapper/OriginateParameters.cs]: Could this be `public int Timeout { get; set; } = 30;`?

**Comment by nufyoot** [comment]
Other than that one comment, LGTM

File: src/AsteriskWrapper/OriginateParameters.cs (hunk 4 above)
Comment (PalletBoy) [inline src/AsteriskWrapper/OriginateParameters.cs]: Didn't know you could do a class dec that way!

**Pull request closed without merging**